session/sql_migration_test.go

```diff
@@ -0,0 +1,738 @@
+			func(t *testing.T, store *BoltStore, _ accounts.Store) {
+				// this test uses caveats & perms from the
+				// tlv_test.go
+				_, err := store.NewSession(
+					ctx, "test", TypeMacaroonAdmin,
+					time.Unix(1000, 0), "foo.bar.baz:1234",
+					WithMacaroonRecipe(caveats, perms),
+				)
+				require.NoError(t, err)
+			},
+		},
+		{
+			"one session with macaroon recipe nil caveats",
+			func(t *testing.T, store *BoltStore, _ accounts.Store) {
+				// this test uses perms from the tlv_test.go
+				_, err := store.NewSession(
+					ctx, "test", TypeMacaroonAdmin,
+					time.Unix(1000, 0), "foo.bar.baz:1234",
+					WithMacaroonRecipe(nil, perms),
+				)
+				require.NoError(t, err)
+			},
+		},
+		{
+			"one session with macaroon recipe nil perms",
+			func(t *testing.T, store *BoltStore, _ accounts.Store) {
+				// this test uses caveats from the tlv_test.go
+				_, err := store.NewSession(
+					ctx, "test", TypeMacaroonAdmin,
+					time.Unix(1000, 0), "foo.bar.baz:1234",
+					WithMacaroonRecipe(caveats, nil),
+				)
+				require.NoError(t, err)
+			},
+		},
+		{
+			"one session with a linked account",
+			func(
+				t *testing.T, store *BoltStore,
+				acctStore accounts.Store,
+			) {
+
+				// Create an account with balance
+				acct, err := acctStore.NewAccount(
+					ctx, 1234, time.Now().Add(time.Hour),
+					"",
+				)
+				require.NoError(t, err)
+				require.False(t, acct.HasExpired())
+
+				// For now, we manually add the account caveat
+				// for bbolt compatibility.
+				accountCaveat := checkers.Condition(
+					macaroons.CondLndCustom,
+					fmt.Sprintf("%s %x",
+						accounts.CondAccount,
+						acct.ID[:],
+					),
+				)
+
+				sessCaveats := []macaroon.Caveat{
+					{
+						Id: []byte(accountCaveat),
+					},
+				}
+
+				_, err = store.NewSession(
+					ctx, "test", TypeMacaroonAccount,
+					time.Unix(1000, 0), "",
+					WithAccount(acct.ID),
+					WithMacaroonRecipe(sessCaveats, nil),
+				)
+				require.NoError(t, err)
+			},
+		},
+		{
+			"linked session",
+			func(t *testing.T, store *BoltStore, _ accounts.Store) {
+				// First create the initial session for the
+				// group.
+				sess1, err := store.NewSession(
+					ctx, "initSession", TypeMacaroonAdmin,
+					time.Unix(1000, 0), "",
+				)
+				require.NoError(t, err)
+
+				// As the store won't allow us to link a
+				// session before all sessions in the group have
+				// been revoked, we revoke the session before
+				// creating a new session that links to the
+				// initial session.
+				err = store.ShiftState(
+					ctx, sess1.ID, StateCreated,
+				)
+				require.NoError(t, err)
+
+				err = store.ShiftState(
+					ctx, sess1.ID, StateRevoked,
+				)
+				require.NoError(t, err)
+
+				_, err = store.NewSession(
+					ctx, "linkedSession", TypeMacaroonAdmin,
+					time.Unix(1000, 0), "",
+					WithLinkedGroupID(&sess1.ID),
+				)
+				require.NoError(t, err)
+			},
+		},
+		{
+			"randomized sessions",
+			randomizedSessions,
+		},
```

Comment:
Opted for a test that uses the "standard" `rand` lib here, as @bitromortac had a preference for that in https://github.com/lightninglabs/lightning-terminal/pull/1047#discussion_r2059882334. But can also look into adding a test that uses `rapid` if you'd prefer.